src/UIComponents/Results.jsx

```diff
@@ -28,59 +28,73 @@ class Results extends Component {
             {this.props.percentDataToReserve}% of the training data was reserved
             to test the accuracy of the newly trained model.
           </p>
+          {isNaN(this.props.summaryStat.stat) && (
+            <p>
+              An accuracy score was not calculated because no training data was
+              reserved for testing.
+            </p>
+          )}
           <div>
-            {this.props.summaryStat.type === MLTypes.REGRESSION && (
-              <div>
+            {this.props.summaryStat.type === MLTypes.REGRESSION &&
+              !isNaN(this.props.summaryStat.stat) && (
                 <div>
-                  The average difference between expected and predicted labels
-                  is:
+                  <div>
+                    The average difference between expected and predicted labels
+                    is:
+                  </div>
+                  <div style={styles.subPanel}>
+                    {this.props.summaryStat.stat}
+                  </div>
                 </div>
-                <div style={styles.subPanel}>{this.props.summaryStat.stat}</div>
-              </div>
-            )}
-            {this.props.summaryStat.type === MLTypes.CLASSIFICATION && (
-              <div>
-                <div style={styles.mediumText}>
-                  The calculated accuracy of this model is:
-                </div>
-                <div style={styles.subPanel}>
-                  {this.props.summaryStat.stat}%
+              )}
+            {this.props.summaryStat.type === MLTypes.CLASSIFICATION &&
+              !isNaN(this.props.summaryStat.stat) && (
```

Comment:
I wonder if some of this kind of logic belongs in redux rather than in the view..